modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/validation/DocumentValidationException.java

```diff
@@ -61,11 +61,17 @@ public DocumentValidationReport getReport() {
     public String getMessage() {
         if (report.hasError()) {
             int num = report.numberOfErrors();
+            ValidationViolation violation = report.asList().get(0);
+            String property = "";
+            if (violation instanceof ConstraintViolation) {
+                ConstraintViolation cv = (ConstraintViolation) violation;
+                property = cv.getPath().get(0).toString();
```

Comment:
`getPath().get(0)` just returns the first part of the property xpath, but for a complex value it would be useful to have the full path. So all parts would be joined by `/`. Note that care must be taken for list elements (see logic in `PathNode.toString()`). All this logic could probably be put in a new static method `PathNode.toString(List<PathNode>)` that returns a `String`.
While doing this please rename `property` to `xpath` as that's what it is now.

Reply:
Can you check if there's a unit test that will show this full xpath in the error message as well?

Reply:
There is not any unit test that show the XPath which is understandable as the XPath wasn't a part of the error message before.

Reply:
Ok so there's probably a need for one to be sure the feature works as we expect.

Reply:
The following test was updated to test the XPath.